ui/src/app/components/rename-modal.tsx

```diff
@@ -55,6 +54,87 @@ export class RenameModal extends React.Component<Props, States> {
     };
   }
 
+  getFileNameWithoutExtension(name: string) {
+    const lastDotIndex = name?.lastIndexOf('.');
+    return lastDotIndex === -1 ? name : name?.slice(0, lastDotIndex);
+  }
+
+  getFileExtension(name: string) {
+    const lastDotIndex = name?.lastIndexOf('.');
+    return lastDotIndex === -1 ? '' : name?.slice(lastDotIndex);
+  }
+
+  validateNewName(
+    existingNames: string[],
+    oldName: string,
+    newName: string,
+    resourceType: ResourceType
+  ) {
+    const { lowerCaseNames, newNameNoExtension } =
+      this.getFilteredListOfFileNamesWithoutExtension(
+        existingNames,
+        oldName,
+        newName
+      );
+
+    const errors = validate(
+      { newName: newNameNoExtension },
+      {
+        newName: nameValidationFormat(lowerCaseNames, resourceType),
+      }
+    );
+    return errors;
+  }
+
+  private getFilteredListOfFileNamesWithoutExtension(
+    existingNames: string[],
+    oldName,
+    newName: string
+  ) {
+    const oldNameExtension = this.getFileExtension(oldName);
+    const oldNameNoExtension = this.getFileNameWithoutExtension(oldName);
+
+    const newNameExtension = this.getFileExtension(newName);
+    let newNameNoExtension = this.getFileNameWithoutExtension(newName || '');
+
+    // Filtering only files that have the same extension as the original file and stripping file extensions from the names
+    const filteredExistingNamesNoExtension = existingNames
+      .filter(
+        (name) =>
+          this.getFileExtension(name).toLowerCase() ===
+          oldNameExtension.toLowerCase()
+      )
+      .map(this.getFileNameWithoutExtension);
+
+    // No need to show error if the new name is the same as the old name but the extension has different case
+    // User could be fixing a typo in the extension
+    // Example: newName.r -> newName.R
+    const isSameNameNoExtension = newNameNoExtension === oldNameNoExtension;
+    const isDifferentNameCaseSensitive = newName ? newName !== oldName : true;
```

Comment:
I want to simplify this by not checking for the existence of `newName` but I'm guessing there's a good reason for this?